File: minimal-editor-api/Enums/Enums.cs
using System.Runtime.Serialization;

public enum Enum_SceneSizeEnum
{
    [EnumMember(Value = "nil")]
    Unset,
    [EnumMember(Value = "s")]
    Small,
    [EnumMember(Value = "m")]
    Medium,
    [EnumMember(Value = "l")]
    Large,
    [EnumMember(Value = "xl")]
    XtraLarge
}

public enum Enum_ActorTypesEnum
{
    [EnumMember(Value = "nil")]
    Unset,
    [EnumMember(Value = "circle")]
    Circle,
    [EnumMember(Value = "rectangle")]
    Rectangle,
    [EnumMember(Value = "background")]
    Background,
    [EnumMember(Value = "resource-bar")]
    ResourceBar,
    [EnumMember(Value = "image")]
    Image
}

File: app/src/enums/enums.ts
export type sizeEnum = 's' | 'm' | 'l' | 'xl';

export const ActorTypeEnum = { // NOTE: Must do like this so that Switch statements of checking work
  rectangle: 'rectangle',
  circle: 'circle',
  resourcebar: 'resource-bar',
  background: 'background',
  image: 'image'
} as const;

export type ActorTypeEnum = typeof ActorTypeEnum[keyof typeof ActorTypeEnum];

export enum MenuItemsEnum {
    NewScene = 1,// "New Scene",
    LoadScene = 2,// "Load Scene",
    SaveScene = 3,// "Save Scene",
    LaunchScene = 4,// "Launch Scene",
    AddActor  = 5,// "Add Actor",
    RemoveActor = 6,// "Remove Actor",
    ModivyActor = 7,// "Modify Actor",
}
